### DISCUSSION TITLE
Автоматическое обновление

### DISCUSSION THREAD

**Discussion by vlalog** [Chimildic/goofy#14] [Идея] [answered]
Дальше мои размышления о том, как это реализовать:
Можно добавить файл version.txt, скопировать ссылку на RAW в библиотеку и периодически проверять. Если версия в файле больше, запрашивать новый файл из гитхаба (или я могу написать api, которое будет вставлять данные с гет параметров в нужные места и возвращать готовый файл) и перезаписывать library.gs

**Comment by Chimildic** [comment, marked as answer]
Некоторые предыдущие обновления требовали дополнительных действий (миграции настроек, обновлении прав доступа). Даже если перезапись самого себя возможна, неизвестно какие действия могут еще потребоваться. А с автоматическим обновлением это означает ломать код работающим триггерам. 
Можно конечно к версии и условный флаг добавить, можно ли автоматически обновлять. Но насколько это нужно вообще?
